hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionProcessor.java

```diff
@@ -60,59 +64,113 @@ public TransactionProcessor(
         this.contractCreation = requireNonNull(contractCreation);
     }
 
+    /**
+     * Process the given transaction, returning the result of running it to completion
+     * and committing to the given updater.
+     *
+     * @param transaction     the transaction to process
+     * @param updater         the world updater to commit to
+     * @param feesOnlyUpdater if base commit fails, a fees-only updater
+     * @param context         the context to use
+     * @param tracer          the tracer to use
+     * @param config          the node configuration
+     * @return the result of running the transaction to completion
+     */
     public HederaEvmTransactionResult processTransaction(
             @NonNull final HederaEvmTransaction transaction,
             @NonNull final HederaWorldUpdater updater,
+            @NonNull final Supplier<HederaWorldUpdater> feesOnlyUpdater,
             @NonNull final HederaEvmContext context,
             @NonNull final HederaTracer tracer,
             @NonNull final Configuration config) {
+        final InvolvedParties parties;
+        final GasCharges gasCharges;
         try {
-            // Compute the sender, relayer, and to address (will throw if invalid)
-            final var parties = computeInvolvedParties(transaction, updater, config);
+            // Compute the sender, relayer, and to address (throws if invalid)
+            parties = computeInvolvedParties(transaction, updater, config);
             if (transaction.isEthereumTransaction()) {
                 parties.sender().incrementNonce();
```

Comment:
Just checking that its correct that the nonce is incremented even if `gasCharging.chargeForGas` throws `HandleException`?

Reply:
We wouldn't call `commit()` in that case, so nothing should happen to state at all.